mlflow/utils/search_utils.py

```diff
@@ -71,22 +92,55 @@ def _get_identifier(cls, identifier):
         return {"type": cls._valid_entity_type(entity_type), "key": key}
 
     @classmethod
-    def _process_token(cls, token):
-        if token.ttype == TokenType.Operator.Comparison:
-            return {"comparator": token.value}
-        elif token.ttype in cls.VALUE_TYPES:
-            return {"value": token.value}
+    def _get_value(cls, identifier_type, token):
+        if identifier_type == cls._METRIC_IDENTIFIER:
+            if token.ttype not in cls.NUMERIC_VALUE_TYPES:
+                raise MlflowException("Expected numeric value type for metric. "
+                                      "Found {}".format(token.value),
+                                      error_code=INVALID_PARAMETER_VALUE)
+            return token.value
         else:
-            return {}
+            if identifier_type != cls._PARAM_IDENTIFIER:
+                # Expected to be either "param" or "metric".
+                raise MlflowException("Invalid identifier type. Expected one of "
+                                      "{}".format([cls._METRIC_IDENTIFIER, cls._PARAM_IDENTIFIER]))
+            if token.ttype in cls.STRING_VALUE_TYPES:
+                return cls._strip_quotes(token.value)
+            elif isinstance(token, Identifier):
+                return cls._strip_quotes(token.value)
+            raise MlflowException("Expected string value type for parameter. "
+                                  "Found {}".format(token.value),
+                                  error_code=INVALID_PARAMETER_VALUE)
+
+    @classmethod
+    def _validate_comparison(cls, tokens):
+        base_error_string = "Invalid comparison clause"
+        if len(tokens) != 3:
+            raise MlflowException("{}. Expected 3 tokens found {}".format(base_error_string,
+                                                                          len(tokens)),
+                                  error_code=INVALID_PARAMETER_VALUE)
+        if not isinstance(tokens[0], Identifier):
+            raise MlflowException("{}. Expected 'Identifier' found '{}'".format(base_error_string,
```

Comment:
Is it possible to put the token values in these error messages?

Reply:
@andrewmchen you are a good code reviewer.

I had a tough time writing tests for these. Everything was getting caught by "needs to be a comparison clause" check. But, now that I think again--I could write test like `1 == 1` or `"LR" == params.model`

Reply:
They are included.